### DISCUSSION TITLE
Enhancement Request

### DISCUSSION THREAD

**Discussion by Volcanicus** [shun126/DungeonGenerator#4] [Ideas]
Hello!

I was wondering if these enhancements could be made:
1. Instead of using static meshes, to use actors for rooms in general. For instance, I have rooms that have special cameras or triggers or checks like "is room cleared?" to drop items.
2. Have rooms be of a specific transform and no grid overlapping. An example would be like in the binding of isaac: rooms are of a certain ratio dimension and their entrances cannot be moved otherwise.

Otherwise your plugin is solid!

**Comment by shun126** [comment]
Thanks for the suggestion.

#1.
It looks like the DungeonRoomSensorActor has a similar function. This class is generated to cover each room. This range can be used to spawn actors and check for clearing.

https://youtube.com/watch?v=qPIFG-ZNLbM
This is explained at 15 minutes 45 seconds into this video.
I hope I have fulfilled your request.

#2.
We want to guarantee that the rooms can be varied in size, so we have not come up with a way to fix the number and location of the gates but not the location of the corridors. We will continue to consider this.
